resources/home/dnanexus/Pan_Cancer_Sample_Report.Rmd

```diff
@@ -313,66 +319,67 @@ The table below lists URL links to different annotation databases. These are:
 
 
 ```{r url_links}
-fusion_abdriged_annotation$gene_fusion1 <- genes[seq_along(genes) %% 2 > 0]
-fusion_abdriged_annotation$gene_fusion2 <- genes[seq_along(genes) %% 2 == 0]
-
-
-make_hyperlink = function(myurl,mytext=myurl) {
-  paste('<a href="',myurl,'">',mytext,'</a>')
-}
-
-mega_fusion_html <-read_html(params$cosmic_fusions)
-my_df <- as.data.frame(mega_fusion_html %>% html_table(fill=TRUE))
-links <- mega_fusion_html %>% html_nodes("a") %>% html_attr("href")
-links_fusions <- grep("https://cancer.sanger.ac.uk/cosmic/fusion/overview?", links, value = TRUE) 
-my_df$links_fusions <- links_fusions
-
-for (i in 1:nrow(my_df)) {
-  fusion <- my_df[i,1]
-  if (grepl("ENST",fusion, fixed = TRUE)) {
-    genes_vector <- str_split_1(fusion, "-")
-    my_df[i,"gene_fusion"] <- paste(gsub("_.*","",genes_vector[1]), gsub("_.*","",genes_vector[2]), sep = "-")
-  } else {
-    my_df[i,"gene_fusion"] <- fusion
+try({
+  fusion_abdriged_annotation$gene_fusion1 <- genes[seq_along(genes) %% 2 > 0]
+  fusion_abdriged_annotation$gene_fusion2 <- genes[seq_along(genes) %% 2 == 0]
+  
+  
+  make_hyperlink = function(myurl,mytext=myurl) {
+    paste('<a href="',myurl,'">',mytext,'</a>')
   }
-}
-
-fusion_abdriged_annotation$gene_nice <- gsub("--", "-", fusion_abdriged_annotation[,1])
-filtered_my_df <- my_df[which(my_df$gene_fusion %in% fusion_abdriged_annotation$gene_nice),]
-fusion_abdriged_annotation$gene_fusion_cosmic <- ""
-fusion_abdriged_annotation$gene_fusion_cosmic <- filtered_my_df$links_fusions[match(fusion_abdriged_annotation$gene_nice, filtered_my_df$gene_fusion)]
-
-fusion_abdriged_annotation$gene_fusion1_oncokb <- c(paste("https://www.oncokb.org/gene/", fusion_abdriged_annotation$gene_fusion1, "/Fusions?refGenome=GRCh38" , sep = ""))
-fusion_abdriged_annotation$gene_fusion2_oncokb <- c(paste("https://www.oncokb.org/gene/", fusion_abdriged_annotation$gene_fusion2, "/Fusions?refGenome=GRCh38" , sep = ""))
-
-fusion_abdriged_annotation$gene_fusion1_CT <- c(paste("https://clinicaltrials.gov/search?term=", fusion_abdriged_annotation$gene_fusion1, "%20Fusion" , sep = ""))
-fusion_abdriged_annotation$gene_fusion2_CT <- c(paste("https://clinicaltrials.gov/search?term=", fusion_abdriged_annotation$gene_fusion2, "%20Fusion" , sep = ""))
```

Comment:
Pls. set all the links as a variable which can be modified outside the code chunks.